sdk/servicebus/azure-servicebus/tests/test_queues.py

```diff
@@ -1337,19 +1338,78 @@ def test_queue_message_settle_through_mgmt_link_due_to_broken_receiver_link(self
                 assert len(messages) == 1
                 messages[0].complete()
 
-    def test_queue_mock_no_reusing_auto_lock_renew(self):
-        class MockReceivedMessage:
-            def __init__(self):
-                self._received_timestamp_utc = utc_now()
-                self.locked_until_utc = self._received_timestamp_utc + timedelta(seconds=10)
 
-            def renew_lock(self):
-                self.locked_until_utc = self.locked_until_utc + timedelta(seconds=10)
+    def test_queue_mock_auto_lock_renew_callback(self):
+        results = []
+        errors = []
+        def callback_mock(renewable, error):
+            results.append(renewable)
+            if error:
+                errors.append(error)
+
+        auto_lock_renew = AutoLockRenew()
+        auto_lock_renew.renew_period = 1 # So we can run the test fast.
+        with auto_lock_renew: # Check that it is called when the object expires for any reason (silent renew failure)
+            message = MockReceivedMessage(prevent_renew_lock=True)
+            auto_lock_renew.register(renewable=message, on_lock_renew_failure=callback_mock)
+            time.sleep(3)
+            assert len(results) and results[-1]._lock_expired == True
+            assert not errors
+
+        auto_lock_renew = AutoLockRenew()
+        auto_lock_renew.renew_period = 1
+        with auto_lock_renew: # Check that in normal operation it does not get called
+            auto_lock_renew.register(renewable=MockReceivedMessage(), on_lock_renew_failure=callback_mock)
+            time.sleep(3)
+            assert len(results) == 1
```

Comment:
I wonder if it's worth resetting results/errors with each scenario - it makes the intentions a bit clearer.